tests/unit/s2n_record_size_test.c

```diff
@@ -444,27 +426,33 @@ int main(int argc, char **argv)
 
         /* Testing with a small blob */
         s2n_stack_blob(small_blob, ONE_BLOCK, ONE_BLOCK);
+        struct iovec small_io_vec = { 0 };
+        small_io_vec.iov_base = small_blob.data;
+        small_io_vec.iov_len = small_blob.size;
 
         int bytes_taken;
 
         const uint16_t TLS13_RECORD_OVERHEAD = 22;
-        EXPECT_SUCCESS(bytes_taken = s2n_record_write(server_conn, TLS_APPLICATION_DATA, &small_blob));
+        EXPECT_SUCCESS(bytes_taken = s2n_record_writev(server_conn, TLS_APPLICATION_DATA, &small_io_vec, 1, 0, small_blob.size));
         EXPECT_EQUAL(bytes_taken, ONE_BLOCK); /* we wrote the full blob size */
         EXPECT_EQUAL(s2n_stuffer_data_available(&server_conn->out), ONE_BLOCK + TLS13_RECORD_OVERHEAD); /* bytes on the wire */
 
         /* Check we get a friendly error if we use s2n_record_write again */
-        EXPECT_FAILURE_WITH_ERRNO(s2n_record_write(server_conn, TLS_APPLICATION_DATA, &small_blob), S2N_ERR_RECORD_STUFFER_NEEDS_DRAINING);
+        EXPECT_ERROR_WITH_ERRNO(s2n_record_write(server_conn, TLS_APPLICATION_DATA, &small_blob), S2N_ERR_RECORD_STUFFER_NEEDS_DRAINING);
         EXPECT_SUCCESS(s2n_stuffer_wipe(&server_conn->out));
-        EXPECT_SUCCESS(s2n_record_write(server_conn, TLS_APPLICATION_DATA, &small_blob));
+        EXPECT_OK(s2n_record_write(server_conn, TLS_APPLICATION_DATA, &small_blob));
         EXPECT_SUCCESS(s2n_stuffer_wipe(&server_conn->out));
 
         /* Testing a big 100k blob to be written */
         s2n_stack_blob(big_blob, ONE_HUNDRED_K, ONE_HUNDRED_K);
+        struct iovec big_io_vec = { 0 };
+        big_io_vec.iov_base = big_blob.data;
+        big_io_vec.iov_len = big_blob.size;
 
         /* Test that s2n_record_write() doesn't error on writing large payloads.
```

Comment:
This comment is no longer correct since we aren't calling `s2n_record_write`.

Also why the switch to `s2n_record_writev`?

Reply:
Same question on 451.